kgraphql/src/test/kotlin/com/apurebase/kgraphql/specification/typesystem/ScalarsSpecificationTest.kt

```diff
@@ -21,6 +22,71 @@ import java.util.UUID
 @Specification("3.1.1 Scalars")
 class ScalarsSpecificationTest {
 
+    @Test
+    fun `built-in scalars should be available by default`() {
+        val schema = KGraphQL.schema {
+            query("int") {
+                resolver<Int> { 1 }
+            }
+            query("float") {
+                resolver<Float> { 2.0f }
+            }
+            query("double") {
+                resolver<Double> { 3.0 }
+            }
+            query("string") {
+                resolver<String> { "foo" }
+            }
+            query("boolean") {
+                resolver<Boolean> { true }
+            }
+            // TODO: ID, cf. https://github.com/stuebingerb/KGraphQL/issues/83
+        }
+
+        val response = deserialize(schema.executeBlocking("{ int float double string boolean }"))
+        assertThat(response.extract<Int>("data/int"), equalTo(1))
+        assertThat(response.extract<Float>("data/float"), equalTo(2.0))
+        assertThat(response.extract<Double>("data/double"), equalTo(3.0))
+        assertThat(response.extract<String>("data/string"), equalTo("foo"))
+        assertThat(response.extract<Boolean>("data/boolean"), equalTo(true))
+    }
+
+    @Test
+    fun `extended scalars should not be available by default`() {
+        invoking {
+            KGraphQL.schema {
+                query("long") {
+                    resolver<Long> { 1L }
+                }
+            }
+        } shouldThrow SchemaException::class withMessage "An Object type must define one or more fields. Found none on type Long"
+
+        invoking {
+            KGraphQL.schema {
+                query("short") {
+                    resolver<Short> { 2.toShort() }
+                }
+            }
+        } shouldThrow SchemaException::class withMessage "An Object type must define one or more fields. Found none on type Short"
+    }
+
+    @Test
+    fun `extended scalars should be available if included`() {
+        val schema = KGraphQL.schema {
+            extendedScalars()
+            query("long") {
+                resolver<Long> { Long.MAX_VALUE }
+            }
+            query("short") {
+                resolver<Short> { 2.toShort() }
+            }
+        }
+
+        val response = deserialize(schema.executeBlocking("{ long short }"))
+        assertThat(response.extract<Long>("data/long"), equalTo(9223372036854775807L))
+        assertThat(response.extract<Int>("data/short"), equalTo(2))
```

Comment:
IMHO this should be `extract<Short>` and `equalTo(2.toShort())` but haven't found out why that doesn't work. There is also a cast in `TestUtils.extract` that apparently successfully casts an `Int` to a `Short`, which shouldn't be possible. Hopefully that will all become clearer during work on #160.